File: calculator/src/main/java/com/turlir/calculator/converter/Visual.java
package com.turlir.calculator.converter;

/**
 * Возможность представления токенов через интерфейс {@link Printer}
 */
public interface Visual {

    /**
     * Посетитель для представления текущего токена
     * @param chain интерфейс вывода на текущем шаге
     */
    void print(Printer chain);

}

File: calculator/src/main/kotlin/com/turlir/calculator/converter/Visual.kt
package com.turlir.calculator.converter

/**
 * Возможность представления токенов через интерфейс [Printer]
 */
interface Visual {

    /**
     * Посетитель для представления текущего токена
     * @param chain интерфейс вывода на текущем шаге
     */
    fun print(chain: Printer)

}
